codeforboston/maple
# Nav-bar Update 2

# Summary

current issue: https://github.com/codeforboston/maple/issues/1389
next issue: https://github.com/codeforboston/maple/issues/1390

- split TopNav into seperate components: DesktopNav and MobileNav
  - this is because the design for each view doesn't use the same order of objects and wanted to avoid tons of nested if statements
- used specific widths for DesktopNav elements because those elements will remain statistically sized across screen sizes;  the Search Element will expand and contract to fit the screen size
  - Currently, there is a Placeholder Widget as we do not have an independent search component at the moment.  This Widget is currently set invisible.  Turn that off if you want to see it but it's sole purpose atm is to take up space.
- profile icon reversed to blue on white, per the Figma
  - matching org icon used where applicable
- Mobile view is minimally altered and resembles the old Hamburger -> Logo -> Profile/Login layout; getting it to match the Figma will be a future PR
- the old "Why Use Maple" dropdown was subsumed into the "Learn" dropdown to preserve spacing

Tertiary Changes:
- One button "browse bills" on the HeroHeader was set to use colors that were hard to distinguish against the background - this was changed to use colors in the Figma
- Log In button on the HeroHeader's text changed to match the Figma

![bluey](https://github.com/codeforboston/maple/assets/73559781/7f88e5ff-7513-47bc-b539-ca8ef36e65c4)

Follow Up Issues/PRs:
- we do not currently have a Universal Search Widget.  I suggest 
  1. potentially using the Bill Search as a temporary placeholder
  2. then comparing Bill Search and Testimony Search to figure out how to put a Universal Search Widget
- update Mobile view per the Figma

[https://www.figma.com/file/IyRpfsVrU2pg3rysLauvQR/[Ready-to-Dev]-Navigation-V2?type=design&node-id=4356-14&mode=design&t=kX9UkJnwrUrn51Tf-0](https://www.figma.com/file/IyRpfsVrU2pg3rysLauvQR/%5BReady-to-Dev%5D-Navigation-V2?type=design&node-id=4356-14&mode=design&t=kX9UkJnwrUrn51Tf-0)

# Checklist

- [n/a ] If I've added shared components, I've added a storybook story.

# Screenshots

![image](https://github.com/codeforboston/maple/assets/73559781/b4c5dbf4-402c-4d5c-b1e7-b5ee6b564310)

![image](https://github.com/codeforboston/maple/assets/73559781/426c4af0-7f3d-4a90-bd89-a6d4c21ee8d7)

![image](https://github.com/codeforboston/maple/assets/73559781/97821236-fa11-41c7-83f7-cb05be251527)

![image](https://github.com/codeforboston/maple/assets/73559781/5cf55d79-2f25-4df9-b7cc-de2460092393)

![image](https://github.com/codeforboston/maple/assets/73559781/a0d07a6f-4848-4151-93f9-24703c099172)

# Known issues

1. hydration issues: used some isClient code in components/layout.tsk to resolve them but there may be a more elegant solution
2. the Figma does not have a tiny arrow next to the profile icon: I can build the dropdown differently to avoid the arrow but then it has subtle inconsistencies when compared to the nearby dropdowns
3. the tiny arrows don't flip up and down when their corresponding dropdowns are clicked on.  I didn't find any Bootstrap functionality to do this.

# Steps to test/reproduce

1. fire up Maple and compare the nav bar to the Figma
2. click on various links to make sure those still work
